src/WebJobs.Script/Metrics/IHostMetrics.cs

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Azure.WebJobs.Script.Metrics
+{
+    /// <summary>
+    /// Defines the host metric meter methods.
+    /// </summary>
+    public interface IHostMetrics
+    {
+        /// <summary>
+        /// Increments the app failure count.
+        /// </summary>
+        public void AppFailure();
```

Comment:
Wondering if the name of this isn't too general - are we trying to capture only host level fatal startup errors? E.g. a failed function invocation is a failure but wouldn't be counted here.

At a minimum, I think you should update this method to detail exactly what an "AppFailure" is. Or perhaps we need to rename it to something more specific like HostStartupFailureCount.

Reply:
At the moment this is for any permenant app failures where customer action is required to fix the issue i.e unable to download app content due to a 404, or the dotnet worker SDK not supported or anytime we encounter a `HostInitializationException` (this is all outlined in the design doc). A future evolution Jacob and I have disuccused is adding a dimension to this for `severity` where critical is for all permenant app failures and error for other types of issues, but that is being figured out by the larger host metrics picture.

If you feel the use cases I mentioned above can be encalsulated by `HostStartupFailureCount` then we can rename it to that, otherwise I can rename to PermentantAppFailures and add more info in the summary

Reply:
Discussed with Fabio and we're going to leave this as is for now. We need to finalize these changes to unblock other components. We can always refactor this once we have our initial iteration working.